### PR TITLE
Lucy/did select place

### PR DESCRIPTION
**Pull request by lucyxubroad**
Addresses #283 Remove the need for didSelectPlace in didSelectRowAt method of AllStopsViewController
- Add initializer method to RouteOptionsViewController so that every instantiation of RouteOptionsViewController must include a "search to" location
- Remove "didSelectPlace" function call to instantiate RouteOptionsViewController from AllStopsViewController

### CODE REVIEW

File: TCAT/Controllers/HomeOptionsCardViewController+Extensions.swift [hunk 1]
```diff
@@ -272,16 +271,20 @@ extension HomeOptionsCardViewController: UITableViewDelegate {
...
             }
         }
 
         tableView.deselectRow(at: indexPath, animated: true)
         searchBar.endEditing(true)
-
+        
         if shouldPushViewController {
-            let vcToPush = didSelectAllStops ? allStopsTableViewConroller : routeOptionsViewController
+            guard let vcToPush = didSelectAllStops ? allStopsTableViewController : routeOptionsViewController else {
```
Comment (Omarrasheed) [inline TCAT/Controllers/HomeOptionsCardViewController+Extensions.swift]: can combine with if
```
if shouldPushViewController, let vcToPush = ... {
  ...
}

**Review by Omarrasheed — APPROVED** [review]
Bet

File: TCAT/Controllers/HomeOptionsCardViewController+Extensions.swift [hunk 2]
```diff
@@ -272,16 +271,20 @@ extension HomeOptionsCardViewController: UITableViewDelegate {
...
+                routeOptionsViewController = RouteOptionsViewController(searchTo: place)
+                routeOptionsViewController.didReceiveCurrentLocation(currentLocation)
                 Global.shared.insertPlace(for: Constants.UserDefaults.recentSearch, place: place)
             }
         }
 
         tableView.deselectRow(at: indexPath, animated: true)
         searchBar.endEditing(true)
-
+
```
Comment (kevinchan159) [inline TCAT/Controllers/HomeOptionsCardViewController+Extensions.swift]: nit: delete whitespace

**Review by kevinchan159 — APPROVED** [review]
a m a z i n g 😍

**Comment by usubzero** [comment]
Looks good :)

**Review by usubzero — APPROVED** [review]

**Pull request merged**